api/v1alpha1/clusterextension_types.go

```diff
@@ -43,6 +43,25 @@ const (
 	UpgradeConstraintPolicyIgnore UpgradeConstraintPolicy = "Ignore"
 )
 
+// Similar to NamespacedName, but with json
+type ClusterExtensionSecretRef struct {
+	// Name of the secret
+	Name string `json:"name"`
+	// Namespace of the secret
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
Should we allow for specification of the namespace here? I'm wondering if using the install namespace as a point where all other resources referenced should exist is more reasonable?

Reply:
The secret is associated with the image registry, not the installation of the bundle itself. This secret is independent of the installation namespace.

Reply:
To be clear, it's not required on every installation.

If the docker-registry's certificate is signed by a system CA, this option is not required. This requires additional setup (i.e. cert-manager/trust-manager/deployment updates).

Reply:
> To be clear, it's not required on every installation.

You are right. But it is required on every installation if you use a registry with self-signed certs (like we do in our own tests: see `addCertToSpec` in this PR). I would expect it to be quite a common pattern.

Reply:
I would actually expect it to be an exception, not a rule. The operator-controller deployment can be modified to include CAs used by the cluster.